### PR TITLE
Feature/add coverage reports

### PR DESCRIPTION
**Pull request by romulocintra**
Added coverage Reports for unit tests

### CODE REVIEW

**Review by desfero — APPROVED** [review]
Thanks a lot

**Comment by filipesilva** [comment]
I think this is a useful feature, but that it shouldn't be the default with running `npm test` since it increases the time it takes to run unit tests.

**Comment by romulocintra** [comment]
Hi @filipesilva  agree with you , and normally on my SPA apps we have a command like  $ npm run validate that runs lint , unit test and coverage ... but in this case if we decide to go for componentization of our projects having more atomic pieces of code , the delay executing test + coverage reports is worthwhile time saving

**Comment by filipesilva** [comment]
It might be worth it in your case, but that's not necessarily true for all projects. What do you think of having a second karma config and a npm script to run coverage using that instead?

**Comment by romulocintra** [comment]
Hi @filipesilva  , do you mean have an specific karma-coverage-config file and add an extra npm command to run coverage !? for me would be ok and yes should be most generic posible in order to fit for all projects .. Start preparing PR

**Comment by filipesilva** [comment]
Yeap, that is what I mean. It would be nice to be able to import and extend the other one, but I'm not sure if there's a good way of doing that.

**Comment by romulocintra** [comment]
Hi @filipesilva , already made a split in Karma conf file ,  we are able to extend it for coverage features , doing a require of general configuration was the most elegant way i found to not replicate to much code .

File: karma-coverage.conf.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  // Generic Karma Configuration
+  karmaConf(config);
+
+  //Extended Configuration for Karma Coverage Reports
+  config.set({
+    plugins: [
+      require('karma-jasmine'),
+      require('karma-chrome-launcher'),
+      require('karma-jasmine-html-reporter'),
+      require('karma-coverage'),
```
Comment (filipesilva) [inline karma-coverage.conf.js:12]: Some of these need to be added to `package.json`.

**Review by filipesilva — CHANGES_REQUESTED** [review]
Great job with the extension!

Could you also add `npm run coverage` to `.travis.yml`? It will test if your config works correctly and make it easier for folks to just copy the CI config.

File: karma-coverage.conf.js (hunk 1 above)
Comment (romulocintra) [reply karma-coverage.conf.js:12]: already added on first PR

**Review by filipesilva — APPROVED** [review]

**Comment by filipesilva** [comment]
Merging now, thanks for all the work you've done in this feature!

**Comment by romulocintra** [comment]
Thanks to you all 👍

**Pull request merged**